packages/frontend/core/src/components/pure/cmdk/data.tsx

```diff
@@ -216,7 +224,25 @@ export const usePageCommands = () => {
   const navigationHelper = useNavigateHelper();
   const t = useAFFiNEI18N();
 
+  const [searchTime, setSearchTime] = useState<number>(0);
+
+  // HACK: blocksuite indexer is async,
+  // so we need to re-search after it has been updated
+  useEffect(() => {
```

Comment:
I think making the `query` a debounced value could also implicitly solve the issue.
Also we should make query debounced to improve the searching performance.

Reply:
The indexer may delay by 1 second,
or the content may not have been synced, which may take longer.

debounce cannot solve this, this is a temporary solution until https://github.com/toeverything/blocksuite/issues/5668 be solved.

Reply:
Oh I just noticed that this function is essentially a setInterval.